### PR TITLE
move third party wallet connections to sdk

### PR DESCRIPTION
**Pull request by hcote**
### 📦 Pull Request

[Provide a general summary of the pull request here.]

- Adds Coinbase Wallet and Wallet Connect dependencies
- Adds a new optional parameter `thirdPartyWalletOptions` to the Magic constructor options (required if using Magic Connect with Coinbase Wallet or Wallet Connect enabled)
- Adds a new `magic.wallet.getProvider()` method that returns the provider of the active wallet
- Moves third party wallet connection logic to the sdk

### ✅ Fixed Issues

- [List any fixed issues here like: Fixes #XXXX]

### 🚨 Test instructions

[Describe any additional context required to test the PR/feature/bug fix.]

### ⚠️ Don't forget to add a [semver](https://semver.org/) label! 
Please only add **one** label:
- `patch`: Bug Fix?
- `minor`: New Feature?
- `major`: Breaking Change?
- `skip-release`: It's unnecessary to publish this change.

### CODE REVIEW

File: packages/@magic-sdk/provider/src/modules/wallet.ts [hunk 1]
```diff
@@ -1,37 +1,278 @@
...
+
+  private getMetaMaskProvider(): any {
+    return (window as any).ethereum?.providers?.find((p: any) => p?.isMetaMask) || (window as any).ethereum;
+  }
+
+  private connectToMetaMask(): Promise<string[]> {
+    // Redirect to MetaMask app if user selects MetaMask on mobile
+    if (this.isMobile() && !this.isMetaMaskInstalled()) {
+      const metaMaskDeepLink = `https://metamask.app.link/dapp/${window.location.href.replace(/(^\w+:|^)\/\//, '')}`;
+      window.location.href = metaMaskDeepLink;
```
Comment (mattupham) [inline packages/@magic-sdk/provider/src/modules/wallet.ts:136]: worth wrapping in `encodeURIComponent` as well?

File: packages/@magic-sdk/provider/test/spec/modules/wallet/connectToCoinbaseWallet.spec.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    },
+  });
+  magic.wallet.request = jest.fn();
+
+  magic.wallet.isMobile = jest.fn(() => true);
+  magic.wallet.isCoinbaseWalletBrowser = jest.fn(() => false);
+
+  await magic.wallet.connectToCoinbaseWallet();
+  expect(window.location.href).toEqual('https://go.cb-w.com/dapp?cb_url=http%3A%2F%2Flocalhost');
+});
```
Comment (mattupham) [inline packages/@magic-sdk/provider/test/spec/modules/wallet/connectToCoinbaseWallet.spec.ts:90]: nice work on these!

**Review by mattupham — APPROVED** [review]

File: scripts/utils/esbuild.ts [hunk 3]
```diff
@@ -41,7 +42,7 @@ export async function build(options: ESBuildOptions) {
         define: Object.fromEntries(
           Object.entries(environment).map(([key, value]) => [`process.env.${key}`, JSON.stringify(value)]),
         ),
-        plugins: [...globalsPlugin(options.globals || {})],
+        plugins: [...globalsPlugin(options.globals || {}), NodeModulesPolyfillPlugin()],
```
Comment (bcleary06) [inline scripts/utils/esbuild.ts:45]: What are we using this for?

Comment (hcote) [reply scripts/utils/esbuild.ts:45]: To fix build errors from the wallet-connect dependency

Screenshot 2023-03-22 at 9 25 44 AM

**Review by mattupham — DISMISSED** [review]

**Review by bcleary06 — DISMISSED** [review]

**Review by mattupham — DISMISSED** [review]

**Review by bcleary06 — APPROVED** [review]

**Comment by magiclabsFE** [comment]
:rocket: PR was released in [`@magic-ext/algorand@9.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/algorand%409.0.0), [`@magic-ext/avalanche@9.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/avalanche%409.0.0), [`@magic-ext/bitcoin@9.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/bitcoin%409.0.0), [`@magic-ext/conflux@7.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/conflux%407.0.0), [`@magic-ext/connect@7.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/connect%407.0.0), [`@magic-ext/cosmos@9.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/cosmos%409.0.0), [`@magic-ext/ed25519@5.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/ed25519%405.0.0), [`@magic-ext/flow@9.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/flow%409.0.0), [`@magic-ext/harmony@9.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/harmony%409.0.0), [`@magic-ext/icon@9.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/icon%409.0.0), [`@magic-ext/near@9.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/near%409.0.0), [`@magic-ext/oauth@8.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/oauth%408.0.0), [`@magic-ext/polkadot@9.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/polkadot%409.0.0), [`@magic-ext/react-native-bare-oauth@9.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/react-native-bare-oauth%409.0.0), [`@magic-ext/react-native-expo-oauth@9.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/react-native-expo-oauth%409.0.0), [`@magic-ext/solana@10.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/solana%4010.0.0), [`@magic-ext/taquito@6.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/taquito%406.0.0), [`@magic-ext/terra@6.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/terra%406.0.0), [`@magic-ext/tezos@9.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/tezos%409.0.0), [`@magic-ext/webauthn@8.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/webauthn%408.0.0), [`@magic-ext/zilliqa@9.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-ext/zilliqa%409.0.0), [`@magic-sdk/commons@10.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/commons%4010.0.0), [`@magic-sdk/pnp@8.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/pnp%408.0.0), [`@magic-sdk/provider@14.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/provider%4014.0.0), [`@magic-sdk/react-native-bare@15.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/react-native-bare%4015.0.0), [`@magic-sdk/react-native-expo@15.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/react-native-expo%4015.0.0), [`@magic-sdk/types@12.0.0`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/types%4012.0.0), [`magic-sdk@14.0.0`](https://github.com/magiclabs/magic-js/releases/tag/magic-sdk%4014.0.0) :rocket:

**Pull request merged**